Topics/Development_Process/GraphQL_VS_REST.md

```diff
@@ -0,0 +1,94 @@
+# GraphQL and REST APIs
+
+## Table of Contents
+1. [Overview & Motivation](#overview--motivation)
+    - [What is an API?](#what-is-an-api)
+    - [Why do we need APIs?](#why-do-we-need-apis)
+    - [Designing APIs](#designing-apis)
+2. [REST Overview](#rest-overview)
+3. [GraphQL Overview](#graphql-overview)
+4. [When does REST make sense to use?](#when-does-rest-make-sense-to-use)
+5. [When does GraphQL make sense to use?](#when-does-graphql-make-sense-to-use)
+6. [Quick Reference](#quick-reference-for-common-uses)
+7. [Summary](#summary)
+8. [Additional Readings](#additional-reading)
+
+## Overview & Motivation
+### What is an API?
+An API is an **application programming interface**, or in simpler words, a way for two pieces of software to communicate with one another.
+
+On this page, we will refer to things called `endpoints`. These are basically access ways for our API - you use these endpoints to communicate with the software.

```

Comment:
I think you could add an example of both an API and an endpoint. A definition/explanation might be a bit vague for some people, whereas an example would make things very clear

Reply:
Great idea! I've added it in a new commit :)